### PR TITLE
[create-block] Add npmDevDependencies as a template variable.

### LINKED ISSUE #39722: Allow devDependencies to be defined for a template
## What problem does this address?
Currently, the template system for `@wordpress/create-block` package allows the definition of `dependencies` but not `devDependencies`. There are many times that a block may need `devDependencies` defined.

## What is your proposed solution?
Add a `npmDevDependencies` variable to the template to align with the existing `npmDependencies`

### PR DESCRIPTION
**Pull request by ryanwelcher**
## What?
This PR adds a new template variable to the create-block package to allow templates to define `devDependencies`.

Closes #39722

## Why?
Some blocks require `devDependences` that should not be added to regular `dependencies`.

## How?
This is done by introducing a `npmDevDependencies` variable

## Testing Instructions
1. Download https://github.com/ryanwelcher/dynamic-block-template and add a `npmDevDependencies` entry to the index.js and add an array of dependencies
Run ./node_modules/.bin/wp-create-block --template={pathTo}/dynamic-block-template

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
I haven't had a chance to test that yet. The changes proposed in the PR look great though 👍🏻 

Any idea why CI jobs didn't start?

**Comment by ryanwelcher** [comment]
Thanks, @gziolo! Not sure about the CI either, I tried pushing another commit to restart the jobs but that did nothing.

File: packages/create-block/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+### New Features
+-    Add `npmDevDependencies` template variable to allow definition `devDependencies`as part of a template ([#39723](https://github.com/WordPress/gutenberg/pull/39723)).
```
Comment (dgwyer) [inline packages/create-block/CHANGELOG.md]:
```suggestion
-    Add `npmDevDependencies` template variable to allow definition of `devDependencies` as part of a template ([#39723](https://github.com/WordPress/gutenberg/pull/39723)).
```

**Review by dgwyer — APPROVED** [review]
Just updated a couple of typos.

**Comment by ryanwelcher** [comment]
For posterity, I was able to get the CI job to re-run by rebasing from `trunk` and force-pushing the changes as per @c4rl0sbr4v0 in WP Make slack.

**Pull request merged**